### PR TITLE
Refactor interval sets

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

File: asmcomp/linscan.ml [hunk 1]
```diff
@@ -41,69 +34,73 @@ type class_intervals =
     mutable ci_fixed: IntervalSet.t;
     mutable ci_active: IntervalSet.t;
     mutable ci_inactive: IntervalSet.t;
-    mutable ci_spilled: SpilledSet.t;
-    mutable ci_free_slots: IntSet.t; (* stack slots available for reuse *)
+    mutable ci_spilled:
+      (* spilled stack slots (reg.loc = Stack (Local n)) still in use *)
+      IntervalSet.t;
```
Comment (nojb) [inline asmcomp/linscan.ml:39]: By using `IntervalSet` instaed of `SpilledSet` we lose the property that we always reuse the smallest possible stack slot, but perhaps we don't care.

**Review by nojb — COMMENTED** [review]
Thanks @gasche for this PR!

I only have one question related to replacing `SpilledSet` by `IntervalSet` (below).

File: asmcomp/linscan.ml (hunk 1 above)
Comment (gasche) [reply asmcomp/linscan.ml:39]: Is that true? I thought that this property was ensured by taking the min element of `SlotSet`, in a different part of the code.

Comment (nojb) [reply asmcomp/linscan.ml:39]: Argh, you are right!

**Pull request merged**